### PR TITLE
Add udf support

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: https://github.com/vmware/go-vcloud-director/issues/474

* Added changes to allow the upload of UDF format ISO files and test to validate that.

### CODE REVIEW

**Review by adambarreiro — APPROVED** [review]
UDF brings old memories of re-writable CDs, I almost forgot it existed! :)

**Review by lvirbalas — APPROVED** [review]
LGTM!

**Review by mikeletux — APPROVED** [review]

**Review by Didainius — APPROVED** [review]

File: govcd/catalog_test.go [hunk 1]
```diff
@@ -963,3 +963,31 @@ func (vcd *TestVCD) Test_CatalogQueryMediaList(check *C) {
 	// Check that media name is what it should be
 	check.Assert(medias[0].Name, Equals, vcd.config.Media.Media)
 }
+
+// Tests System function UploadMediaImage by checking if provided UDF type standard iso file uploaded.
```
Comment (lvirbalas) [inline govcd/catalog_test.go:967]: > if provided UDF type standard iso file uploaded.

This part of the sentence doesn't make sense. Can you fix it on the way in some of your open PRs?

Comment (vbauzysvmware) [reply govcd/catalog_test.go:967]: UDF type of ISO file uploaded?

File: govcd/catalog_test.go [hunk 2]
```diff
@@ -963,3 +963,31 @@ func (vcd *TestVCD) Test_CatalogQueryMediaList(check *C) {
...
+func (vcd *TestVCD) Test_CatalogUploadMediaImageWihUdfTypeIso(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	if vcd.config.Media.MediaUdfTypePath == "" {
+		check.Skip("Skipping test because no UDF type iso path given")
```
Comment (lvirbalas) [inline govcd/catalog_test.go:972]: Another message that doesn't make sense. Please fix in some other open PR.
(Why this PR was merged without my approval? ;) )

File: govcd/catalog_test.go (hunk 1 above)
Comment (lvirbalas) [reply govcd/catalog_test.go:967]: We agreed on:

> // Tests System function UploadMediaImage by using provided ISO file of UDF type.

While in commit:

>  // Tests System function UploadMediaImage by checking if provided ISO file of UDF type.

Comment (vbauzysvmware) [reply govcd/catalog_test.go:967]: @mikeletux updated his PR.

**Pull request merged**